### ISSUE TITLE
Drag/drop files from desktop into app

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#2287]
@jmockler @peller @childsb @jhpedemonte 

Right now, if a user wants to use his own images with Maqetta, he has to upload those files to the cloud and then use the files from the Files palette. Modern browsers now allow drag/drop from local system onto browsers. Here is a web page that shows programming techniques for using new HTML5 features in this area:

http://www.html5rocks.com/en/tutorials/file/dndfiles/

The question now is where in the app should we allow dropping files? My first thought is:
- Allow dropping arbitrary files into Files palette, which will trigger an auto-upload of the file. (I'm not sure if the drag/drop features in HTML5 provide the original file name. If not, we may need to post a dialog asking user to supply a filename. Note that we might need a small server enhancement to do some base64 unencoding.
- Allow dropping arbitrary images onto the canvas. In this case, we could just use data protocol and base64 encoding of the image, but my instincts are that we should prompt user whether to add a new file to workspace or use "inline image (data: protocol)"
- (Extra credit) Drop onto background-image field in properties palette

**Comment by jhpedemonte** [comment]
> Allow dropping arbitrary images onto the canvas. In this case, we could just use data protocol and base64 encoding of the image, but my instincts are that we should prompt user whether to add a new file to workspace or use "inline image (data: protocol)"

Not sure if we want to use the data: protocol at all.  Just prompt user for where to save image in project, then add image widget with correct URL to page.

**Comment by peller** [comment]
Agreed.  No choices on DnD sounds like a better user experience.  If we really want to support the data protocol, perhaps that would be a separate operation, available to all images, but probably best left to some other optimization tool.
